test/conftest.py

```diff
@@ -156,26 +156,10 @@ def rag_generator():
 
 
 @pytest.fixture()
-def faiss_document_store():
```

Comment:
@lalitpagaria You introduced this in https://github.com/deepset-ai/haystack/pull/484/commits/874b51b6762ee121e0f0ebb37161316c2ece86ac . Can you please describe again your reasoning? I think it was related to lowering the memory consumption, but can't recall the details ...

Reply:
Ideally with each marker no test should fail because of dependency. For example if we only want to test `test_db.py` then we don't need `xpdf` or `tika` and tests should not fail event these dependencies are not there. This I done intentionally so developer can test small changes by running small set of tests only.

Reply:
But I like the changes done by @tanaysoni. These are more clean. 
So we can add back my changes again we need them in future.

Only please take care of `document_store.faiss_index.reset()` for faiss doc store.